src/Datasource/EntityTrait.php

```diff
@@ -241,7 +248,11 @@ public function set(array|string $field, mixed $value = null, array $options = [
         if (!is_array($field)) {
             throw new InvalidArgumentException('Cannot set an empty field');
         }
-        $options += ['setter' => true, 'guard' => $guard];
+        $options += ['setter' => true, 'guard' => $guard, 'asOriginal' => false];
```

Comment:
We should update the doc blocks for this method as well. I can take care of that if you don't want to.

Reply:
Done. Would you take a look if it makes sense?